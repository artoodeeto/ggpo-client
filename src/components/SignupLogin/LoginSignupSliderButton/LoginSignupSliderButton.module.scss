.SliderBtn {
  font-size: 1em;
}
