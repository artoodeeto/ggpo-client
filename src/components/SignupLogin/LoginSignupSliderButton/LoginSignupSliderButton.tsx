import React, { FC, Fragment } from 'react';
import styles from './LoginSignupSliderButton.module.scss';

type LoginSignupSliderButtonProps = {
  toShow: (val: boolean) => void;
};

export const LoginSignupSliderButton: FC<LoginSignupSliderButtonProps> = ({ toShow }) => {
  return (
    <Fragment>
      <button className={styles.SliderBtn} type="button" onClick={() => toShow(true)}>
        Already a User?
      </button>
      <button className={styles.SliderBtn} type="button" onClick={() => toShow(false)}>
        New User
      </button>
    </Fragment>
  );
};
