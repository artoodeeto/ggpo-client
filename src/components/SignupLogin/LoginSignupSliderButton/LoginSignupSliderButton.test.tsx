/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { LoginSignupSliderButton } from './LoginSignupSliderButton';

describe('<LoginSignupSliderButton/>', () => {
  describe('snapshot', () => {
    it('', () => {
      const wrapper = shallow(<LoginSignupSliderButton toShow={(val) => {}} />);
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('button tags', () => {
    const wrapper = shallow(<LoginSignupSliderButton toShow={(val) => {}} />);
    it('should count 2 buttons', () => {
      expect(wrapper.find('button').length).toBe(2);
    });
    it('should have class name of sliderBtn', () => {
      expect(wrapper.find('.SliderBtn')).toHaveClassName('SliderBtn');
      expect(wrapper.find('.SliderBtn').length).toBe(2);
    });
    it('should have text', () => {
      expect(wrapper.find('button').at(0).text()).toEqual('Already a User?');
      expect(wrapper.find('button').at(1).text()).toEqual('New User');
    });
    it('onClick should have been called', () => {
      const spyFunc = jest.fn(() => true);
      const cont = shallow(<LoginSignupSliderButton toShow={spyFunc} />);
      cont.find('button').at(0).simulate('click');
      cont.find('button').at(1).simulate('click');
      expect(spyFunc).toHaveBeenCalledTimes(2);
      spyFunc.mockRestore();
    });
  });
});
