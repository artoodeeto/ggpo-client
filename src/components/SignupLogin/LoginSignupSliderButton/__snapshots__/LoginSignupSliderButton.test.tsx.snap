// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginSignupSliderButton/> snapshot  1`] = `
<Fragment>
  <button
    className="SliderBtn"
    onClick={[Function]}
    type="button"
  >
    Already a User?
  </button>
  <button
    className="SliderBtn"
    onClick={[Function]}
    type="button"
  >
    New User
  </button>
</Fragment>
`;
